lib/target_mvd/targets.ex

```diff
@@ -47,9 +47,20 @@ defmodule TargetMvd.Targets do
   end
 
   def create_target(attrs \\ %{}) do
-    %Target{}
-    |> Target.changeset(attrs)
-    |> Repo.insert()
+    user_id = extract_option(:user_id, attrs)
+
+    targets =
+      Target
+      |> where(user_id: ^user_id)
+      |> Repo.aggregate(:count, :id)
+
+    if targets < 10 do
```

Comment:
Is this something `doable` with Plug (not just possible but something that you want to delegate to)? WDYT @brunvez @mcousillas6? 🤔

Reply:
```
defmodule App.TargetController do
  use App.Web, :controller

  plug App.ValidationPlug when action in [:create]
end
```

Something like that, not sure if that works 🤔

Reply:
Take a look at [this post](https://www.amberbit.com/blog/2019/6/29/refactoring-phoenix-controllers/)

Reply:
I think this is a context concern, `Plug` is part of the web layer of your app while this part here does not depend on it. Maybe the post talks more about checking parameters and data sanity. I'd say anything that needs the DB should be on a context